### PR TITLE
fiosync: a tool to sync an ostree repo across factories

### PR DESCRIPTION
**Pull request by mike-sul**
- Pull an ostree repo commit from a source factory by using `ostree`
command line utility.
- Push the pulled ostree commit/repo to a destination factory by using
existing mechanism/Pusher, the same that is used by CI builds.

Signed-off-by: Mike Sul <mike.sul@foundries.io>

### CODE REVIEW

File: pkg/ostreeuploader/pull.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	if err := p.ostreehub.auth(); err != nil {
+		return err
+	}
+	url := *p.ostreehub.url
+	url.Path = path.Join(url.Path, "download-urls")
+	req, err := http.NewRequest("POST", url.String(), nil)
+	if err != nil {
+		log.Fatalf("Failed to create a request to get download URLs: %s\n", err.Error())
+	}
+	req.Header.Set("X-Correlation-ID", "corId")
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply pkg/ostreeuploader/pull.go:69]: yep, I had a fever when I was writing this code.

File: pkg/ostreeuploader/pull.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to get download URLs: %s", resp.Status)
+	}
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply pkg/ostreeuploader/pull.go:94]: makes sense.

**Comment by mike-sul** [comment]
@doanac force pushed changes according to the comments.

**Comment by doanac** [comment]
> Unfortunately, it takes some time for ostree to pull a commit (the push part is quite fast).

its no faster or slower than it always was. i suppose you could make it factory specific and download the ostree tarball from CI. However, I think this is good for now.

**Review by doanac — APPROVED** [review]

**Comment by mike-sul** [comment]
> i suppose you could make it factory specific and download the ostree tarball from CI.

But, it won't work for customers that don't use our CI.

**Pull request merged**